riotboot_dfu: select DFU_ALT automatically when invoking make flash-slotX

dylad opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions.
-->

### Contribution description
This PR adds an automatic selection for DFU_ALT when invoking `make flash-slotX`. This way, we don't have to manually set it (as I forget it every single time). Update the documentation accordingly.


### Testing procedure

`PROGRAMMER_QUIET=0 PROGRAMMER=dfu-util USEMODULE="usbus_dfu"  make BOARD=saml21-xpro -C tests/buttons riotboot/flash-slot1` and checks DFU_ALT variable in your terminal log.
`PROGRAMMER_QUIET=0 PROGRAMMER=dfu-util USEMODULE="usbus_dfu"  make BOARD=saml21-xpro -C tests/buttons riotboot/flash-slot0` and checks again your DFU_ALT variable.
Everytime, the flash-slotX should match DFU_ALT=X.

<!--
Details steps to test your contribution:
- which test/example to compile for which board and is there a 'test' command
- how to know that it was not working/available in master
- the expected success test output
-->


### Issues/PRs references

None

chrysn commented on bootloaders/riotboot_dfu/doc.txt:
```diff
@@ -55,8 +55,10 @@ $ FEATURES_REQUIRED+=riotboot USEMODULE+=usbus_dfu make -C examples/saul BOARD=p
 Note that when building and flashing a different slot (eg. `flash-slot1`),
-the DFU utility has to be explicitly prompted to upload it the other slot
-by adding a `DFU_ALT=1` argument.
+the DFU utility need to be explicitly prompted to upload it the other slot
```
Now that this is automated, is there much point in still having this section? Then again, we could emphasize what changes:

```suggestion
not only is the image built for that slot, but also dfu-util gets passed `--alt 1` (via the `DFU_ALT` build variable) to store it in the right place.
```

(Is there any trick to make multi-line suggestions? I'm afrait this can't just be click-accepted.)

chrysn reviewed (approved):
Does the right thing both from code inspection and testing.

If you want to take a form of the suggestion, feel free to force-push that right away.

dylad commented:
@chrysn 
I've updated the PR and squashed at the same time.
(note that you can select multiple lines for suggestions by holding your left click on the lines you want, you'll see a yellow vertical bar near the line number to show all the selected line then click on the suggestion button).

chrysn commented:
Tools are a bit picky about comit message lengths; can you strip the second down? Suggestion:     "bootloaders/riotboot_dfu: update doc now that DFU_ALT gets set"
